nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -1,112 +1,99 @@
-define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'util/string'], (
-    StagingServiceClient,
-    html,
-    Runtime,
-    UI,
-    String
-) => {
+define([
+    'jquery',
+    'bluebird',
+    'StagingServiceClient',
+    'common/cellComponents/paramsWidget',
+    'common/html',
+    'common/runtime',
+    'common/props',
+    'common/spec',
+    'common/ui',
+    'util/string',
+], ($, Promise, StagingServiceClient, ParamsWidget, html, Runtime, Props, Spec, UI, String) => {
     'use strict';
     const div = html.tag('div'),
         p = html.tag('p'),
         aTag = html.tag('a'),
         ul = html.tag('ul'),
         li = html.tag('li'),
-        select = html.tag('select'),
-        option = html.tag('option'),
-        input = html.tag('input'),
-        label = html.tag('label'),
         fieldset = html.tag('fieldset'),
         cssBaseClass = 'kb-xsv-gen';
 
     const DOCS_LINK = 'https://docs.kbase.us/data/upload-download-guide/bulk';
 
-    const formConfig = {
-        layout: [
-            'types', // multiple choice
-            'output_file_type', // CSV / TSV / EXCEL
-            'output_directory', // file path
-        ],
-        specs: {
-            types: {
-                original: {
-                    advanced: 0,
-                    allow_multiple: 1,
-                    default_values: [
-                        '', // current type
+    const xsvAppSpec = {
+        parameters: [
+            {
+                advanced: 0,
+                allow_multiple: 0,
+                default_values: [
+                    '', // current type
+                ],
+                description: 'File types to generate templates for',
+                disabled: 0,
+                dropdown_options: {
+                    multiselection: 1,
+                    options: [
+                        // map file type to display name
```

Comment:
this gets populated when the class is instantiated using the types from the bulk import cell spec